Programs/_testembed.c

```diff
@@ -1760,8 +1760,10 @@ static int check_use_frozen_modules(const char *rawval)
 
 static int test_init_use_frozen_modules(void)
 {
+    // bpo-45020: Test that -X "frozen_modules" works.
     const char *envvar = getenv("TESTFROZEN");
-    return check_use_frozen_modules(envvar);
+    wchar_t *frozen = Py_DecodeLocale(envvar, NULL);
```

Comment:
If you are going to use `Py_DecodeLocale()`, you need to check its result for failure, and free the dynamic allocated memory after use. You should also add a special case for `getenv()` returning NULL.

@Fidget-Spinner's suggestion is much simpler.